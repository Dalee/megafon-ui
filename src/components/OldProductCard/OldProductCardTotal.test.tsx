import * as React from 'react';
import { shallow } from 'enzyme';
import OldProductCardTotal from './OldProductCardTotal';

const props = {
    payment: {
        value: '600 ₽',
        oldValue: '750 ₽',
        unit: 'в месяц',
    },
};

describe('<OldProductCardTotal />', () => {
    it('it renders OldProductCardTotal', () => {
        const wrapper = shallow(<OldProductCardTotal {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<OldProductCardTotal />', () => {
    it('it handle handleSubmit prop', () => {
        const handleSubmit = jest.fn();
        const wrapper = shallow(<OldProductCardTotal {...props} onSubmit={handleSubmit} />);

        wrapper.find('.old-product-card-total__button').simulate('click', {} as React.SyntheticEvent);
        expect(handleSubmit).toHaveBeenCalledTimes(1);
    });
});
