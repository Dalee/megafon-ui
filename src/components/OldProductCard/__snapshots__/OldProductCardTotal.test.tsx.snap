// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OldProductCardTotal /> it renders OldProductCardTotal 1`] = `
<div
  className="old-product-card-total"
>
  <div
    className="old-product-card-total__wrap"
  >
    <div
      className="old-product-card-total__cost old-product-card-total__cost_old-value"
    >
      <span
        className="old-product-card-total__price"
      >
        600 ₽
      </span>
       
      <span
        className="old-product-card-total__old-price"
      >
        750 ₽
      </span>
      <br />
      <span
        className="old-product-card-total__unit"
      >
        в месяц
      </span>
    </div>
    <div
      className="old-product-card-total__buttons"
    />
  </div>
</div>
`;
